### PR TITLE
[feat] shorten filename for uses and hierarchy

### PR DESCRIPTION
**Pull request by bomgar**

### CODE REVIEW

**Comment by bomgar** [comment]
see https://github.com/ensime/ensime-emacs/issues/656

**Comment by bomgar** [comment]
I'm not sure if `project-roots` is the correct method here

File: ensime-editor.el [hunk 1]
```diff
@@ -897,11 +897,15 @@ falls back to the classic version."
...
 (defun ensime-format-source-position (source-position)
   "Format source position SOURCE-POSITION."
-    (let* ((file-name (ensime-pos-file source-position))
-           (maybe-line (ensime-pos-line source-position)))
-      (let ((line (if maybe-line (number-to-string (if (= 0 maybe-line) 1 maybe-line)) "?")))
-       (concat file-name
-       (propertize (concat ":" line) 'face 'font-lock-comment-face)))))
+  (let* ((file-name (ensime-pos-file source-position))
+         (maybe-line (ensime-pos-line source-position))
+         (roots (if (fboundp 'project-roots)(project-roots (project-current)) '()))
```
Comment (fommil) [inline ensime-editor.el]: you could the `:root-dir` from the `.ensime`

**Comment by bomgar** [comment]
`ensime-configured-project-root` works. thanks.

**Comment by bomgar** [comment]
Should I change something else or is this good to go?

**Comment by fommil** [comment]
oh sorry I didn't realise you'd pushed, no need to squash anymore I can do it on merge :smile: 

awesome stuff, looking forward to using it.

**Comment by fommil** [comment]
@bomgar I forgot about this and upgraded and when I got results back today I was like "woah, what happened here? This is cool!" :smile:

**Comment by fommil** [comment]
the hierarchy summary seems to be showing the FQN that is referenced now rather than the code snippet. Was that intentional? It is actually very useful. Or did I just not notice before because of the noise?

**Comment by bomgar** [comment]
That was intentional. The server api does not return a preview for the hierarchy.

**Comment by fommil** [comment]
Very nice.

**Pull request merged**